paymentsheet/src/main/java/com/stripe/android/link/ui/wallet/WalletViewModel.kt

```diff
@@ -255,8 +256,22 @@ internal class WalletViewModel @Inject constructor(
     private suspend fun performPaymentConfirmation(
         selectedPaymentDetails: ConsumerPaymentDetails.PaymentDetails,
     ) {
-        // Check if billing details are missing before proceeding with payment
-        if (!selectedPaymentDetails.supports(configuration.billingDetailsCollectionConfiguration, linkAccount)) {
+        val effectivePaymentDetails = selectedPaymentDetails.withEffectiveBillingDetails(configuration, linkAccount)
+        // Check if effective billing details satisfy requirements
+        if (effectivePaymentDetails.supports(configuration.billingDetailsCollectionConfiguration, linkAccount)) {
+            // Use the cached phone for this payment detail if available (ie the user updated it locally)
+            val effectivePhone = linkAccountManager.consumerState.value
+                ?.paymentDetails?.find { it.details.id == effectivePaymentDetails.id }
+                ?.billingPhone ?: linkAccount.unredactedPhoneNumber
+
+            val linkPaymentMethodDetails = LinkPaymentMethod.ConsumerPaymentDetails(
+                details = effectivePaymentDetails,
+                collectedCvc = cvcController.formFieldValue.value.takeIf { it.isComplete }?.value,
+                billingPhone = effectivePhone
+            )
+            performPaymentConfirmationWithDetails(linkPaymentMethodDetails)
```

Comment:
To match iOS, we'll use the effective billing details to confirm the payment if they're valid, rather than always presenting the billing details confirmation flow.